factoryWithTypeCheckers.js

```diff
@@ -379,6 +381,52 @@ module.exports = function(isValidElement, throwOnDirectAccess) {
     return createChainableTypeChecker(validate);
   }
 
+  function createSetTypeChecker(expectedType) {
+    function validate(props, propName, componentName, location, propFullName, secret) {
+      var propValue = props[propName];
+      if (!(propValue instanceof Set)) {
```

Comment:
indeed, this should module-level-cache the getter for Set.prototype.size, call-bind it, and invoke it here.